db/rdbms/src/main/java/io/camunda/db/rdbms/sql/ProcessInstanceMapper.java

```diff
@@ -21,4 +23,65 @@ public interface ProcessInstanceMapper {
   Long count(ProcessInstanceDbQuery filter);
 
   List<ProcessInstanceEntity> search(ProcessInstanceDbQuery filter);
+
+  enum ProcessInstanceSearchColumn implements SearchColumn<ProcessInstanceEntity> {
+    PROCESS_INSTANCE_KEY("key", ProcessInstanceEntity::key),
+    PROCESS_DEFINITION_KEY("processDefinitionKey", ProcessInstanceEntity::processDefinitionKey),
+    PROCESS_DEFINITION_ID("bpmnProcessId", ProcessInstanceEntity::bpmnProcessId),
+    PROCESS_DEFINITION_NAME("processName", ProcessInstanceEntity::processName),
+    PROCESS_DEFINITION_VERSION("processVersion", ProcessInstanceEntity::processVersion),
+    PROCESS_DEFINITION_VERSION_TAG("processVersionTag", ProcessInstanceEntity::processVersionTag),
+    START_DATE("startDate", ProcessInstanceEntity::startDate, DateUtil::fuzzyToOffsetDateTime),
+    END_DATE("endDate", ProcessInstanceEntity::endDate, DateUtil::fuzzyToOffsetDateTime),
```

Comment:
❓Aren't these just `OffsetDateTime`s? What's the reason for this util?

Reply:
Nope. In JSON payloads all dates are technically Strings and need to be parsed. And the searchAfter/searchBefore Object[] array will never get parsed before. ES/OS seems to handle this automatically.